src/tds/statement_methods.cr

```diff
@@ -35,55 +35,74 @@ module TDS
-          when value = scanner.scan(/--.*/)              # single line comment
-          when comment_start = scanner.scan(/\/\*/)      # multi-line comment
-            if comment_end = scanner.scan(/.*?\*\//m)
-              value = comment_start + comment_end
-            else
-              value = comment_start + scanner.rest
+          when value = scanner.scan(/\[([^\]]|\]\])*\]/m) # bracket quoted value, including escaped right brackets such as `]]`
+            # treat as a literal and do not modify
+          when value = scanner.scan(/'([^']|'')*'/m) # single quoted value, including escaped single-quotes such as `''`
+            # treat as a literal and do not modify
+          when value = scanner.scan(/"([^"]|"")*"/m) # double quoted value, including escaped double-quotes such as `""`
+            # treat as a literal and do not modify
+          when value = scanner.scan(/--.*/) # single line comment
+            # minify statement by removing comments
+            value = ""
+          when comment_start = scanner.scan(/\/\*/) # multi-line comment
+            unless scanner.scan(/.*?\*\//m)
               scanner.terminate
             end
+            # minify statement by removing comments
+            value = ""
           when unnamed_param = scanner.scan('?') # unnamed parameter placeholder
             index += 1
-            value = "@PARAM_CRYSTAL_TDS_$#{index}__"
+            value = "@P$#{index}__"
             parameters << value
             unnamed_params_found = true
           when named_param = scanner.scan(/\$\d+/) # named parameter placeholder
             if names.has_key? named_param
-              value = names[named_param] # named param already declared, so we can use the saved param name for it
+              # named param already declared, so we can use the saved param name for it
+              value = names[named_param]
             else
-              idx = named_param[1..].to_i # use the index in the named param to find the related argument to bind to
-              value = "@PARAM_CRYSTAL_TDS_#{named_param}__"
+              # use the index in the named param to find the related argument to bind to
+              idx = named_param[1..].to_i
+              value = "@P#{named_param}__"
               parameters << value
-              names[named_param] = value                # save the named param in case its reused later in the query
-              reordered_arguments << arguments[idx - 1] # argument values need to be reordered to match declaration order
+              # save the named param in case its reused later in the query
+              names[named_param] = value
+              # argument values need to be reordered to match declaration order
+              reordered_arguments << arguments[idx - 1]
             end
             named_params_found = true
+          when value = scanner.scan(/\s+/m) # one or more whitespace characters
```

Comment:
Replacing multiple newlines with a single line break might produce confusing error messages where the line number doesn't match.
An error message like "Syntax error in line 34" will no longer match the erroneous line.
But this will only happen in rare cases.

Reply:
That's a good point, I've just added a new commit which preserves all newlines so line numbers in error messages continue to make sense